service/controller/app/v1/resource/configmap/desired.go

```diff
@@ -77,7 +77,12 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 	return configMap, nil
 }
 
-func (r *Resource) getConfigMap(ctx context.Context, configMapName, configMapNamespace string) (*corev1.ConfigMap, error) {
+func (r *Resource) getConfigMap(ctx context.Context, configMapName, configMapNamespace string) (map[string]string, error) {
+	if configMapName == "" {
```

Comment:
Why can this be an empty string?

Reply:
All levels of config in the app CR are optional. So it will be empty if no config has been set. 

I think it's better to check once here and return early. Rather than checking separately for catalog, app and user config.

Reply:
👍